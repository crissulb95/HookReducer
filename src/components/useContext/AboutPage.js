import React, { useContext } from 'react'
import { UserContext } from './userContext';

export const AboutPage = () => {

    const { user, setUser } = useContext(UserContext);
    console.log(user);

    const handleClick = () => {
        setUser({});
    }

    return (
        <div>
            <h2>AboutPage</h2>
            <hr />

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>

            <button 
                className='btn btn-warning'
                onClick={ handleClick }
            >
                Cerrar sesi√≥n
            </button>
        </div>
    )
}
