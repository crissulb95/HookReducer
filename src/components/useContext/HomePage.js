import React, { useContext } from 'react'
import { UserContext } from './userContext'

export const HomePage = () => {

    const { user, setUser } = useContext(UserContext);
    console.log(user);

    return (
        <div>
            <h2>HomePage</h2>
            <hr />

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>

            <button 
                className='btn btn-outline-warning'
                onClick={ () => setUser({}) }
            >
                Cerrar sesi√≥n
            </button>
        </div>
    )
}
