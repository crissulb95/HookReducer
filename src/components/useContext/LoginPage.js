import React, { useContext } from 'react'
import { UserContext } from './userContext'

export const LoginPage = () => {

    const { user, setUser } = useContext(UserContext);

    return (
        <div>
            <h2>LoginPage</h2>
            <hr />
            <button 
                className='btn btn-outline-primary'
                onClick={ () => setUser({ ...user, id: 1, name: 'Harry' }) }
            >
                Iniciar sesi√≥n
            </button>
        </div>
    )
}
