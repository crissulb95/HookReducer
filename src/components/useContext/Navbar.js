import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">

            <Link className="navbar-brand" to="/">Aplicaci√≥n</Link>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                <NavLink activeClassName='active' className="nav-item nav-link" exact to='/'> Home </NavLink>
                <NavLink activeClassName='active' className="nav-item nav-link" exact to='/about'> About </NavLink>
                <NavLink activeClassName='active' className="nav-item nav-link" exact to='/login'> Login </NavLink>
                </div>
            </div>

        </nav>
    )
}
