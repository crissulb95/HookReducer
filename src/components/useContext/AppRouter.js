import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";

import { Navbar } from './Navbar';

import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <Navbar />

                <div className='container pt-4'>
                    <Switch>
                        <Route path='/about' exact component={ AboutPage } />
                        <Route path='/login' exact component={ LoginPage } />
                        <Route path='/' exact component={ HomePage } />
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
