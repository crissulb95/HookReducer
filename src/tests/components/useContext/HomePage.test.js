import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomePage } from '../../../components/useContext/HomePage';
import { UserContext } from '../../../components/useContext/userContext';

describe('Pruebas en HomePage', () => {

    const user = {
        id:1,
        name : 'Cristian',
        email: 'correo@correo.com'
    }

    const setUser = () => {}

    const wrapper = mount( //shallow solo renderea el componente principal
        //mount deberia hacer el snapshot correctamente pero hay problemas de compatibilidad con react17
    <UserContext.Provider value={{
        user, setUser
    }}>
        <HomePage />
    </UserContext.Provider>
    
    );
    
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    

});
