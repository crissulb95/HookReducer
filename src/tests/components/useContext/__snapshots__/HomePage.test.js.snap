// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomePage Debe mostrarse correctamente 1`] = `<HomePage />`;
